app/lib/efile/md/md502_calculator.rb

```diff
@@ -255,7 +273,108 @@ def calculate_line_d_count_total
       def calculate_line_d_amount_total
         # Add line A, B and C amounts
         line_or_zero(:MD502_LINE_A_AMOUNT) + line_or_zero(:MD502_LINE_B_AMOUNT) + line_or_zero(:MD502_LINE_C_AMOUNT)
+      end
+
+      def calculate_line_1e
+        total_interest = @direct_file_data.fed_taxable_income + @direct_file_data.fed_tax_exempt_interest
+        total_interest > 11_600
+      end
+
+      def calculate_line_7; end
+
+      def calculate_line_15; end
+
+      def calculate_line_16; end
+
+      FILING_MINIMUMS_NON_SENIOR = {
+        single: 14_600,
+        dependent: 14_600,
+        married_filing_jointly: 29_200,
+        married_filing_separately: 14_600,
+        head_of_household: 21_900,
+        qualifying_widow: 29_200
+      }
+
+      FILING_MINIMUMS_SENIOR = {
+        single: 16_550,
+        dependent: 16_550,
+        married_filing_jointly: 30_750,
+        married_filing_separately: 14_600,
+        head_of_household: 23_850,
+        qualifying_widow: 30_750
+      }
+
+      def calculate_deduction_method
+        gross_income_amount = @intake.tax_calculator.gross_income_amount
+        filing_minimum = if @intake.primary_senior? && @intake.spouse_senior? && @intake.filing_status_mfj?
+                           32_300
+                         elsif @intake.primary_senior?
+                           FILING_MINIMUMS_SENIOR[@intake.filing_status]
+                         else
+                           FILING_MINIMUMS_NON_SENIOR[@intake.filing_status]
+                         end
+        if gross_income_amount >= filing_minimum
+          "S"
+        else
+          "N"
+        end
+      end
+
+      DEDUCTION_TABLES = {
+        s_mfs_d: {
+          12000 => 1_800,
+          17999 => ->(x) { x * 0.15 },
+          18000 => 2_700,
+        },
+        mfj_hoh_qss: {
+          24333 => 3_650,
+          36332 => ->(x) { x * 0.15 },
+          36333 => 5_450,
+        }
+      }.freeze
+      FILING_STATUS_GROUPS = {
+        s_mfs_d: [:single, :married_filing_separately, :dependent],
+        mfj_hoh_qss: [:married_filing_jointly, :head_of_household, :qualifying_widow]
+      }.freeze
+
+      def calculate_line_17
+        if @lines[:MD502_DEDUCTION_METHOD]&.value == "S"
+          status_group_key = FILING_STATUS_GROUPS.find { |_, group| group.include?(@intake.filing_status) }[0]
+          deduction_table = DEDUCTION_TABLES[status_group_key]
+          md_agi = line_or_zero(:MD502_LINE_16)
+          amount_or_method = deduction_table.find { |agi_limit, _| md_agi <= agi_limit }[1]
+          if amount_or_method.is_a?(Proc)
+            amount_or_method.call(md_agi)
+          else
+            amount_or_method
```

Comment:
[dust] nice. a little confusing, but I get it. do you think having methods that return hardcoded values, e.g. `12000 => -> { 1_800 }` would be better so that it can always be called? or is that also a bit confusing.

Reply:
i think i don't quite understand the suggestion? the method needs to multiply the agi by 0.15 so unless we predict every possible AGI i'm not sure how that would work. but also maybe you're saying something else

Reply:
yeah I'm saying like in the deduction tables have:
```
        s_mfs_d: {
          12000 => -> { 1_800 },
          17999 => ->(x) { x * 0.15 },
          18000 => -> { 2_700 },
        },
```
so you can reduce this to 
```suggestion
          deduction_method = deduction_table.find { |agi_limit, _| md_agi <= agi_limit }[1]
          deduction_method.call(md_agi)
```

but idk is that's better or not

Reply:
ohhh i understand. yeah i also don't know if that's better